wpe/src/main/cpp/Browser/WPECookieManager.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (C) 2022 Igalia S.L. <info@igalia.com>
+ *   Author: Jani Hautakangas <jani@igalia.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "WPECookieManager.h"
+
+#include <wpe/webkit.h>
+
+#include "Browser.h"
+#include "JNI/JNI.h"
+#include "Logging.h"
+
+DECLARE_JNI_CLASS_SIGNATURE(JNIWPECookieManagerCallback, "com/wpe/wpeview/WPECookieManager$Callback");
+DECLARE_JNI_CLASS_SIGNATURE(JNIWPECookieManager, "com/wpe/wpeview/WPECookieManager");
+
+namespace {
+
+enum class CookieAcceptPolicy {
+    AcceptAlways,
+    AcceptNever,
+    AcceptNoThirdParty
+};
+
+class WPECookieManagerCallbackCache final : public JNI::TypedClass<JNIWPECookieManagerCallback> {
+public:
+    WPECookieManagerCallbackCache()
+        : JNI::TypedClass<JNIWPECookieManagerCallback>(true)
+        , m_onResult(getStaticMethod<void(JNIWPECookieManagerCallback, jboolean)>("onNativeResult"))
+    {
+    }
+
+    void onResult(JNIWPECookieManagerCallback callback, jboolean result) const noexcept
+    {
+        try {
+            m_onResult.invoke(callback, result);
```

Comment:
m_commitResult.invoke(callback, result);

Reply:
You can call it however you want, I put that here because it is how I called it in my example, to help following my proposal. But feel free to use the name you want. The important point here is not the name, but the `@FunctionalInterface` usage.

Reply:
fixed